lib/bundle/brew_installer.rb

```diff
@@ -97,29 +97,34 @@ def service_change_state!(verbose:)
       end
     end
 
-    def link_change_state!(verbose: false, force: false)
-      case @link
+    def link_change_state!(verbose: false)
+      link_args = []
+      link_args << "--force" if unlinked_and_keg_only?
+
+      cmd = case @link
+      when :overwrite
+        link_args << "--overwrite"
+        "link" unless linked?
       when true
-        unless linked_and_keg_only?
-          puts "Force-linking #{@name} formula." if verbose
```

Comment:
This seems like a behaviour, maybe messaging, change? Can you detail the expected changes in behaviour/messaging? Thanks!

Reply:
Here goes, let me know if there's something you think I miss.

In relation to the changes in the `case` statement logic so it checks against `linked?`:
1. There were two cases where it was possible a second run of `brew bundle` would recall `brew link` or `brew unlink`. First gave a Warning, second a full on-disk run with zero changes e.g.

a.  
```
Using gnupg
Force-linking gnupg formula.
Warning: Already linked: /usr/local/Cellar/gnupg/2.4.5_1
To relink, run:
  brew unlink gnupg && brew link gnupg
```

b. 
```
Using gnupg@2.2
Unlinking gnupg@2.2 formula.
Unlinking /usr/local/Cellar/gnupg@2.2/2.2.44... 0 symlinks removed.
```
2. This behaviour now cannot occur.

The verbose output changes to be like how output is done on install. The only existing output that will change is that "`Force-linking <@formula_name> formula`" becomes "`Linking <@formula_name> formula with "--force"`".

A couple of changes in relation to `force` that have no user visible effect:
1. `brew help link` suggests `--force` is to "Allow keg-only formulae to be linked." Deleted line 103 called `--force` on both keg-only and not-keg-only. 
a. Now only add`--force` as an argument in the one case it is relevant, an unlinked keg-only keg.
b. Even though it was a no-op in the not-keg-only case, this refinement is cleaner expression of intent.
4. Can drop `force:` from the parameters list as it is redundant for linking:
a. The original global flag would have no effect unless you had also set `link: true` against a formula in your `Brewfile` 
b. As we can assume that if you explicitly set `link: true` against a formula in your `Brewfile` then you definitely want linking done we can consider the keg-only need for `--force` to be implied.

Reply:
Thanks for explaining. This seems to make sense to me 👍🏻